### PR TITLE
httpbakery: add assert argument to CheckRequest

### PR DESCRIPTION
**Pull request by rogpeppe**
Also change it to return the information declared.

This is the first support for "declared" outside the checkers package.

### CODE REVIEW

File: bakery/example/idservice/idservice.go [hunk 1]
```diff
@@ -351,19 +351,19 @@ type context struct {
 	req *http.Request
 }
 
-func (ctxt *context) CheckFirstPartyCaveat(caveat string) error {
-	op, rest, err := checkers.ParseCaveat(caveat)
-	if err != nil {
-		return fmt.Errorf("cannot parse caveat %q: %v", caveat, err)
-	}
-	switch op {
+func (ctxt *context) Condition() string {
```
Comment (mattyw) [inline bakery/example/idservice/idservice.go:354]: Exported functions should have comments

File: bakery/example/idservice/idservice.go [hunk 2]
```diff
@@ -351,19 +351,19 @@ type context struct {
...
+	return ""
+}
+
+func (ctxt *context) Check(cond, arg string) error {
```
Comment (mattyw) [inline bakery/example/idservice/idservice.go:358]: comment here as well please

**Comment by mhilton** [comment]
:+1: LGTM

**Comment by jrwren** [comment]
:+1:

**Comment by rogpeppe** [comment]
@mattyw sorry, forgot to add the comments. Will address later as in a rush now!

**Pull request merged**